test/net_http_test.rb

```diff
@@ -233,6 +233,11 @@
       assert_equal "foo bar", tempfile.meta["etag"]
     end
 
+    it "only normalizes URLs when URI says the URL is invalid" do
+      Addressable::URI.expects(:parse).never
+      Down::NetHttp.download("#{$httpbin}/etag/2ELk8hUpTC2wqJ%2BZ%25GfTFA.jpg")
+    end
```

Comment:
I would prefer if we could test actual behaviour rather than relying on implementation. The issue was that Addressable modifies the URL into something else. So I would like the test to assert that the request made was to the original URL, instead of the different URL that Addressable's normalization would produce.

Reply:
I couldn't come up with a better way to test it, but that definitely makes sense. I updated the test so it's testing the `base_uri` of the tempfile.